.travis.yml

```diff
@@ -9,48 +9,68 @@ cache:
-          env: TOXENV=py35
-        - python: 3.6
-          env: TOXENV=py36
-        - python: 3.6
-          env: TOXENV=py36_cython
-        - python: 3.7
-          env: TOXENV=py37
-        - python: 3.8
-          env: TOXENV=py38
-        - python: 3.8
-          env: TOXENV=py38_cython
-        - python: 3.8
-          # NOTE(vytas): Big-endian architecture
-          arch: s390x
-          os: linux
-          env: TOXENV=py38_cython
-        - python: 3.8
-          env: TOXENV=py38_smoke
-        - python: 3.8
-          env: TOXENV=py38_smoke_cython
-        - python: 3.8
-          env: TOXENV=docs
-        # TODO(kgriffs): Re-enable once hug has a chance to address
-        #   breaking changes in Falcon 3.0
+        # - python: pypy3.6-7.1.1
+        #   env: TOXENV=pypy3
+        # - python: 3.8
+        #   env: TOXENV=pep8
+        # - python: 3.8
+        #   env: TOXENV=pep8-examples
+        # - python: 3.8
+        #   env: TOXENV=mypy
+        #   # NOTE(kgriffs):  3.5.2 is the default Python 3 version on Ubuntu 16.04
+        #   #   so we pin to that for testing to make sure we are working around
+        #   #   and quirks that were fixed in later micro versions.
+        # - python: 3.5.2
+        #   env: TOXENV=py35
+        # - python: 3.6
+        #   env: TOXENV=py36
+        # - python: 3.6
+        #   env: TOXENV=py36_cython
         # - python: 3.7
-        #   env: TOXENV=hug
-        - python: 3.8
-          env: TOXENV=look
-        - python: 3.8
-          env: TOXENV=check_vendored
+        #   env: TOXENV=py37
+        # - python: 3.8
+        #   env: TOXENV=py38
+        # - python: 3.8
+        #   env: TOXENV=py38_cython
+        # - python: 3.8
+        #   # NOTE(vytas): Big-endian architecture
+        #   arch: s390x
+        #   os: linux
+        #   env: TOXENV=py38_cython
+        # - python: 3.8
+        #   env: TOXENV=py38_smoke
+        # - python: 3.8
+        #   env: TOXENV=py38_smoke_cython
+        # - python: 3.8
+        #   env: TOXENV=docs
+        # # TODO(kgriffs): Re-enable once hug has a chance to address
+        # #   breaking changes in Falcon 3.0
+        # # - python: 3.7
+        # #   env: TOXENV=hug
+        # - python: 3.8
+        #   env: TOXENV=look
+        # - python: 3.8
+        #   env: TOXENV=check_vendored
+
+        - name: Python 3.8 (Windows)
+          env: TOXENV=py38_nocover
+          os: windows
+          language: bash
+          before_install:
+            - choco install python3 --params "/InstallDir:C:\\Python"
+            - export PATH="/c/Python:/c/Python/Scripts:$PATH"
+            - python -m pip install --upgrade pip wheel
+
+          # TODO: Figure out why Travis CI fails the build when a spawned
+          #   uvicorn process does not terminate cleanly and/or figure out
+          #   a way to cleanly shut down the uvicorn process in the first
+          #   place.
+          script: tox -- -k 'not test_asgi_servers'
```

Comment:
I guess we could move forward with enabling the gate like this until we sort out the root cause?

Reply:
The whole behaviour is very strange on windows. The docs seem to say something, but what works is something else thats not supposed to work.

I'm not sure if it's uvicorn that does something strange that makes the process behave this way or it's simply buggy the python implementation on windows. (Maybe it's just windows that's biggy)